### ISSUE TITLE
Prevent adding redirect to yourself in sieve

### ISSUE THREAD

**Issue by Borgso** [roundcube/roundcubemail#8846]
Prevent user from adding redirect to itself.
This is causing a loop in postfix if user does.

status=bounced (mail forwarding loop for USER@DOMAIN.TLD)

**Comment by dilyanpalauzov** [comment]
This is a task for the ManageSieve server.

**Comment by alecpl** [comment]
I agree, RFC 5228: "Implementations MUST take measures to implement loop control, possibly including adding headers to the message or counting Received headers.  If an implementation detects a loop, it causes an error.".

Still, we could maybe do this in the client, but then define "itself". Is this the user's username, or the default identity? Definitely not all defined identities. I wonder whether such limitation could cause any issues.

**Comment by valarauco** [comment]
Hi! 
We are experiencing this same "problem". In our case, Dovecot deals very well with the loop:
`redirect action: not forwarding message to <user@domain>: the `x-sieve-redirected-from' header indicates a mail loop`

But it would be nice if RC didn't allow forwarding emails to an account that is the same of my mailbox (¿IMAP login?), or the mailbox in which the sieve-rule operates. 
Some other email services requests for verification of the forward recipient. (Do I have permission to forward emails to this other address?)

This may be difficult to implement, but I just wanted to put some ideas to this topic because we are in similar situation.
